scanner.go

```diff
@@ -202,13 +216,26 @@ func (s *scanner) request() (*pb.ScanResponse, hrpc.RegionInfo, error) {
 	)
 
 	if s.isRegionScannerClosed() {
+		// filter out the renew option since this is not for renewing
+		// we do this because the response when setting renew=true does
+		// not create a new Scan
+		options := s.rpc.Options()
+		filteredOptions := make([]func(hrpc.Call) error, 0, len(options))
+		for _, opt := range options {
+			dummyScan := &hrpc.Scan{}
+			opt(dummyScan)
+			if !dummyScan.Renew() {
+				filteredOptions = append(filteredOptions, opt)
+			}
+		}
+
```

Comment:
Hopefully not needed after implementing suggestion above.

Reply:
Removed now!